pwndbg/commands/strings.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+import argparse
+from typing import List
+import pwndbg
+import pwndbg.commands
+import pwndbg.aglib.memory
+from pwndbg.commands import CommandCategory
+
+import re
+
+from pwndbg.lib.memory import Page
+
+parser = argparse.ArgumentParser(description="Extracts and displays ASCII strings from readable memory pages of the debugged process.")
+
+parser.add_argument(
+    "-n", type=int, default=4, help="Minimum length of ASCII strings to include")
+parser.add_argument(
+    "page_names", type=str, nargs="*", default=[], help="Mapping to search [e.g. libc].\nCan be used with multiple mappings [e.g libc heap stack]")
+parser.add_argument(
+    "--save-as", type=str, default=None, help="Sets the filename for the output of this command [e.g. --save-as='out.txt']")
+
+@pwndbg.commands.ArgparsedCommand(parser, category=CommandCategory.LINUX)
+@pwndbg.commands.OnlyWhenRunning
+def strings(n: int=4, page_names: List[str]=[], save_as: str=None):
+    # Extract pages with PROT_READ permission
+    readable_pages: List[Page] = [page for page in pwndbg.aglib.vmmap.get() if page.read]
```

Comment:
I guess this could just be a generator expression instead of list comprehension, or, even inlined in line 28, but its good to go for now.